examples/content_management_examples/server/examples/todos/todos.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  ContentStorage,
+  StorageContext,
+  ContentManagementServerSetup,
+} from '@kbn/content-management-plugin/server';
+import { v4 } from 'uuid';
+import {
+  createInSchema,
+  searchInSchema,
+  Todo,
+  TODO_CONTENT_ID,
+  updateInSchema,
+  TodoSearchOut,
+  TodoCreateOut,
+  TodoUpdateOut,
+  TodoDeleteOut,
+  TodoGetOut,
+  createOutSchema,
+  getOutSchema,
+  updateOutSchema,
+  searchOutSchema,
+} from '../../../common/examples/todos';
+
+export const registerTodoContentType = ({
+  contentManagement,
+}: {
+  contentManagement: ContentManagementServerSetup;
+}) => {
+  contentManagement.register({
+    id: TODO_CONTENT_ID,
+    schemas: {
+      content: {
+        create: {
+          in: {
+            data: createInSchema,
+          },
+          out: {
+            result: createOutSchema,
+          },
+        },
+        update: {
+          in: {
+            data: updateInSchema,
+          },
+          out: {
+            result: updateOutSchema,
+          },
+        },
+        search: {
+          in: {
+            query: searchInSchema,
+          },
+          out: {
+            result: searchOutSchema,
+          },
+        },
+        get: {
+          // TODO: why no get in ?
+          out: {
+            result: getOutSchema,
+          },
+        },
+
+        // TODO: why no delete?
+      },
+    },
+    storage: new TodosStorage(),
+  });
+};
+
+class TodosStorage implements ContentStorage {
+  private db: Map<string, Todo> = new Map();
+
+  constructor() {
+    const id1 = v4();
+    this.db.set(id1, {
+      id: id1,
+      title: 'Learn Elasticsearch',
+      completed: true,
+    });
+    const id2 = v4();
+    this.db.set(id2, {
+      id: id2,
+      title: 'Learn Kibana',
+      completed: false,
+    });
+  }
+
+  // TODO: how to connect this to TodoGetIn?
+  // TODO: hot to connect this to input and output schema?
```

Comment:
Again: see that there is no connection to schema at this point

Reply:
`TodoGetOut` is the interface at the RPC level. It is a different one from what the storage layer returns. Here I would assume something like `Promise<Todo>` as a response. It is controlled by the team and it is the content, _not_ the transport layer.

You can see here. https://github.com/elastic/kibana/blob/main/src/plugins/content_management/server/core/crud.ts#L11 that what is returned from this storage `get()` method will be wrapped inside an object under in `item` prop. Those different objects are what @vadimkibana mentioned at some point.